### PR TITLE
wip support compressed files

### PR DESCRIPTION
**Pull request by universalmind303**
So i still have a lot of cleanup on this to do. But in summary, the compressed formats weren't working with our scan providers, even after adding in the support. I was still getting some errors with the encoding. So I started looking in to what datafusion already provides. In doing this, I realized almost all of our object store table providers (with the exception of http) can be replaced with datafusions providers via the ListingTable. 

There's a bit of finessing to get them wired up correctly, but I think going this route can dramatically simplify parts of the codebase & give us much less code to maintain.

### CODE REVIEW

File: crates/glaredb/src/local.rs [hunk 1]
```diff
@@ -79,8 +91,6 @@ impl LocalClientOpts {
                 "\\mode MODE",
                 "Set the output mode [table, json, ndjson, csv]",
             ),
-            ("\\max-rows NUM", "Max number of rows to display"),
-            ("\\max-columns NUM", "Max number of columns to display"),
```
Comment (scsmithr) [inline crates/glaredb/src/local.rs:83]: We still want these.

File: crates/glaredb/src/local.rs [hunk 2]
```diff
@@ -296,11 +308,11 @@ impl LocalSession {
                 println!("Created new session. New database path: {path}");
                 *self = new_sess;
             }
-            ("\\quit", None) => return Ok(ClientCommandResult::Exit),
+            ("\\quit", None) => std::process::exit(0),
```
Comment (scsmithr) [inline crates/glaredb/src/local.rs:311]: This prevents history from being saved, hence the `ClientCommandResult`

File: crates/glaredb/src/local.rs [hunk 3]
```diff
@@ -284,8 +298,6 @@ impl LocalSession {
                 self.opts.mode = OutputMode::from_str(val, true)
                     .map_err(|s| anyhow!("Unable to set output mode: {s}"))?;
             }
-            ("\\max-rows", Some(val)) => self.opts.max_rows = Some(val.parse()?),
-            ("\\max-columns", Some(val)) => self.opts.max_rows = Some(val.parse()?),
```
Comment (scsmithr) [inline crates/glaredb/src/local.rs:288]: Still want these.

File: crates/glaredb/src/local.rs (hunk 2 above)
Comment (universalmind303) [reply crates/glaredb/src/local.rs:311]: will update this file. Was a bad merge.

**Comment by universalmind303** [comment]
most of the sources are working now. a few left to resolve
| Scheme | Glob | Compression | Complete |
|--------|------|-------------|----------|
| HTTP   |      |             | ✅        |
| HTTP   | ✔️   |             | N/A        |
| HTTP   |      | ✔️           | ✅        |
| HTTP   | ✔️   | ✔️           | N/A        |
| File   |      |             | ✅        |
| File   | ✔️   |             | ✅        |
| File   |      | ✔️           | ✅        |
| File   | ✔️   | ✔️           | ✅        |
| GS     |      |             | ✅        |
| GS     | ✔️   |             | ✅        |
| GS     |      | ✔️           | ✅         |
| GS     | ✔️   | ✔️           | ❌        |
| S3     |      |             | ❌       |
| S3     | ✔️   |             | ❌        |
| S3     |      | ✔️           | ❌        |
| S3     | ✔️   | ✔️           | ❌        |

**Comment by scsmithr** [comment]
Pinging @vrongmeal.

There's changes here super familiar with, will need help on making sure this works for us.

**Comment by universalmind303** [comment]
@vrongmeal I can't figure out why the parquet readers aren't working anymore? it seems like they are trying to do some extra work that the csv & json readers arent. Did you run into this with the original implementation?

**Comment by vrongmeal** [comment]
I did remove the old parquet reader factory since we were using the default registry. Didn't face any issues with it. What's the issue you're facing right now?

**Comment by vrongmeal** [comment]
I am guessing, introducing a custom registry breaks the parquet stuff. We did configure parquet reader because of the same.

**Comment by universalmind303** [comment]
closing in favor of #1490

**Pull request closed without merging**